src/ft/mod.rs

```diff
@@ -390,7 +391,7 @@ impl FreeTypeRasterizer {
             return Ok(glyph.font_key);
         }
 
-        for fallback_font in &fallback_list.list {
+        for fallback_font in &fallback_list.list.clone() {
```

Comment:
Cloning is not an option here. This shouldn't be necessary.

Reply:
The problem was that the list over which this loop iterates is borrowed immutably from self, while in the loop the `face_from_pattern` call borrows self mutably. It used to be no problem, because the loop always ended after the call to `face_from_pattern'.

To fix it, I did some refactoring. I have inlined the `face_from_pattern` call. Because that causes code duplication, I moved as much of the non-mutating parts as possible to a new function. Since this new function basically just creates a new FaceLoadingProperties, I placed it in FaceLoadingProperties.